src/TextArea.js

```diff
@@ -0,0 +1,246 @@
+import propTypes from '@dhis2/prop-types'
+import React, { PureComponent } from 'react'
+import cx from 'classnames'
+
+import { statusPropType } from './common-prop-types.js'
+import { StatusIconNoDefault } from './icons/Status.js'
+import { detectScrollbarSize } from './Tabs/detectScrollbarSize.js'
+
+import { styles } from './TextArea/styles.js'
+
+export class TextArea extends PureComponent {
+    constructor(props) {
+        super(props)
+
+        this.textareaRef = React.createRef()
+
+        this.state = {
+            hasScrollbar: false,
+            height: 'auto',
+        }
+
+        this.scrollbarSize = detectScrollbarSize()
+        this.textareaDimensions = { width: 0, height: 0 }
+        this.userHasResized = false
+
+        this.setTextareaDimensions = this.setTextareaDimensions.bind(this)
```

Comment:
Why not just declaring the method like this:

```js
setTextareaDimensions = () => { ... }
```

Then you wouldn't have to bind functions in the constructor anymore and when reading through the method I know immediately that it's bound to the classes' instance.